src/AwkwardPythonCallExt.jl

```diff
@@ -0,0 +1,116 @@
+module AwkwardPythonCallExt
+using PythonCall
+using JSON
+import AwkwardArray
+
+function AwkwardArray.convert(layout::AwkwardArray.Content)::Py
+    form, len, containers = AwkwardArray.to_buffers(layout)
+
+    py_buffers = Dict{String,Any}()
+
+    for (key, buffer) in containers
+        py_buffers[key] = pyimport("numpy").asarray(buffer, dtype = pyimport("numpy").uint8)
+    end
+
+    pyimport("awkward").from_buffers(form, len, py_buffers)
+end
+
+function AwkwardArray.convert(array::Py)::AwkwardArray.Content
+    form, len, _containers = pyimport("awkward").to_buffers(array)
+    containers = pyconvert(Dict, _containers)
+
+    julia_buffers = Dict{String,Vector{UInt8}}()
+
+    for (key, buffer) in containers
+        julia_buffers[key] = reinterpret(UInt8, buffer)
+    end
+
+    AwkwardArray.from_buffers(
+        pyconvert(String, form.to_json()),
+        pyconvert(Int, len),
+        julia_buffers,
+    )
+end
+
+# rule functions
+function pyconvert_rule_awkward_array_primitive(::Type{AwkwardArray.PrimitiveArray}, x::Py)
```

Comment:
`func` is called as `func(S, x::Py)` for some `S <: T`.
The target type `S` is never a union or the empty type, i.e. it is always a data type or union-all. Note, `AwkwardArray.XXXArray` is a `UnionAll` Type.

Reply:
@Moelf, it might be a problem that `AwkwardArray.*Array` are showing up as [UnionAll](https://docs.julialang.org/en/v1/devdocs/types/#UnionAll-types):

> This is needed to describe parametric types where the values of some parameters are not known.

These arrays are supposed to be fully parameterized. At least, that was the intent.

(I acknowledge that this is not something that can be fixed easily or in this PR. I just want to understand it better, to know if Julia is telling us, by its UnionAll type, that our data structures are not as parameterized as we want them to be.)

Reply:
yes, this is a problem, the root cause is the same problem as I pointed out in https://github.com/JuliaHEP/AwkwardArray.jl/issues/40, basically our types are a little messed up right now, need non-trivial fix.

Reply:
That could be the biggest thing to fix, now that the Python conversions are done.